lfs.c

```diff
@@ -8,6 +8,23 @@
 #include "lfs.h"
 #include "lfs_util.h"
 
+// Configuration Sanity Check
+#if (LFS_NAME_MAX <= 0) || (LFS_NAME_MAX > 1022)
+#error "LFS_NAME_MAX must be in the range (0, 1022]"
+#endif
+
+#if (LFS_FILE_MAX <= 0) || (LFS_FILE_MAX > 4294967296)
+#error "LFS_FILE_MAX must be in the range (0, 4294967296]"
+#endif
+
+#if (LFS_FILE_MAX > 2147483647)
+#warning "LFS_FILE_MAX>2147483647; lfs_file_seek, lfs_file_size, and lfs_file_tell will not function properly."
```

Comment:
This is only valid in [C23](https://en.cppreference.com/w/c/preprocessor/error) and compiler-dependent before that.

Also, given that some people opt to use `-Werror` for compilation, this unintentionally breaks their builds.

Reply:
I was a little hesitant on this as well; not sure how frequently users change `LFS_FILE_MAX`, would be open to alternative proposals (or even just fully removing this warning).

Reply:
Curiously gcc does not warn about this, usually `gcc -Werror -std=c99 -Wall -Wextra -pedantic` warns about c99 compat issues. Otherwise CI would have caught this.
https://github.com/littlefs-project/littlefs/actions/runs/6685719140/job/18164313418?pr=886#step:7:50

I wonder why this is...

But @BenBE is right, there's already enough requests for C89 support, we need to stick to C99.

---

Alternative proposal: we can just remove the $&gt;2^{31}$ support. I think realistically no one is using this. It's already only partially supported and untested.

It's a desperate attempt to stretch out what 32-bits can do, but I already have some plans to eventually add a 64-bit mode, and that will probably be a better way to support larger file sizes when needed.

[Worst case](https://xkcd.com/1172/), if someone does depend on this we can always revert.

Reply:
I'm in favor of removing `>2**31` support for the reasons you outlined.